test/schema_test.exs

```diff
@@ -138,4 +139,60 @@ defmodule OpenApiSpex.SchemaTest do
     assert :ok = Schema.validate(schema, "bar", %{})
   end
 
+  test "Validate schema type object when value is array" do
+    schema = %Schema{
+      type: :object
+    }
+    assert {:error, _} = Schema.validate(schema, [], %{})
+  end
+
+  test "Validate schema type array when value is object" do
+    schema = %Schema{
+      type: :array
+    }
+    assert {:error, _} = Schema.validate(schema, %{}, %{})
+  end
+
+  test "Validate schema type boolean when value is object" do
+    schema = %Schema{
+      type: :boolean
+    }
+    assert {:error, _} = Schema.validate(schema, %{}, %{})
+  end
+
+  test "Validate schema type string when value is object" do
+    schema = %Schema{
+      type: :string
+    }
+    assert {:error, _} = Schema.validate(schema, %{}, %{})
+  end
+
+  test "Validate schema type string when value is DateTime" do
+    schema = %Schema{
+      type: :string
+    }
+    assert :ok = Schema.validate(schema, DateTime.utc_now(), %{})
+  end
+
+  test "Validate schema type object when value is DateTime" do
+    schema = %Schema{
+      type: :object
+    }
+    assert :ok = Schema.validate(schema, DateTime.utc_now(), %{})
```

Comment:
mbuhot 6 hours ago  Collaborator
Should this be valid? I would expect DateTimes to serialize to / from strings only.

  @tapickell
tapickell an hour ago
This passes because the DateTime is a Map.
I can see what I can do to get this to error.
ATM I just wanted to document the current behavior with regards to DateTime